apps/src/p5lab/AnimationPicker/AnimationPickerListItem.jsx

```diff
@@ -21,62 +22,93 @@ class AnimationPickerListItem extends React.Component {
   };
 
   state = {
-    loaded: false
+    loaded: false,
+    hover: false
   };
 
   render() {
-    const rootStyle = [styles.root, !this.props.label && styles.noLabel];
+    const {
+      icon,
+      animationProps,
+      category,
+      onClick,
+      playAnimations,
+      label
+    } = this.props;
+    const {loaded, hover} = this.state;
+    const rootStyle = [styles.root, !label && styles.noLabel];
 
     const thumbnailStyle = [
       styles.thumbnail,
-      this.props.icon && styles.thumbnailIcon,
-      this.props.animationProps && {
-        display: this.state.loaded ? 'block' : 'none'
+      icon && styles.thumbnailIcon,
+      animationProps && {
+        display: loaded ? 'block' : 'none'
       }
     ];
 
     const labelStyle = [
       styles.label,
-      this.props.icon && styles.labelIcon,
-      this.props.animationProps && {
-        display: this.state.loaded ? 'block' : 'none'
+      icon && styles.labelIcon,
+      animationProps && {
+        display: loaded ? 'block' : 'none'
       }
     ];
-    const iconImageSrc = this.props.category
-      ? `/blockly/media/p5lab/animation-previews/category_${
-          this.props.category
-        }.png`
+    const iconImageSrc = category
+      ? `/blockly/media/p5lab/animation-previews/category_${category}.png`
       : '';
 
+    const centerStyle = {
+      top: THUMBNAIL_SIZE / 2 - 12,
+      left: THUMBNAIL_SIZE / 2 - 12
+    };
+
+    const hoverIcon = [styles.hoverIcon, centerStyle];
+
+    const hoverBorder = {
+      borderStyle: 'solid',
+      borderRadius: 12,
+      cursor: 'pointer',
+      borderColor: color.purple,
+      borderWidth: '3px',
+      padding: 0
+    };
+
+    const displayMultiSelect = experiments.isEnabled(experiments.MULTISELECT);
+
+    const thumbnailStyleWithHover = [thumbnailStyle, hover && hoverBorder];
+
     return (
       <div
         style={rootStyle}
-        onClick={this.props.onClick}
+        onClick={onClick}
         className="uitest-animation-picker-item"
+        onMouseEnter={() => this.setState({hover: true})}
+        onMouseLeave={() => this.setState({hover: false})}
```

Comment:
Just curious, is there a preference to handle hover state styling this way (using state and in the component) vs. adding a css pseudo class for `:hover`? I'm not sure how we handle our styling and if there are any patterns we follow.

Reply:
oh, maybe I just realized the answer to my question 😅  Is it because we want to maintain that state to show the hover icon?

Reply:
Yup! That's the thinking behind it. :)